masonry/src/widget/portal.rs

```diff
@@ -141,6 +141,31 @@ impl<W: Widget> Portal<W> {
             false
         }
     }
+
+    // Note - Rect is in child coordinates
+    // TODO - Merge with pan_viewport_to
+    // Right now these functions are just different enough to be a pain to merge.
+    pub fn pan_viewport_to_raw(
+        &mut self,
+        portal_size: Size,
+        content_size: Size,
+        target: Rect,
+    ) -> bool {
+        let viewport = Rect::from_origin_size(self.viewport_pos, portal_size);
+
+        let new_pos_x = compute_pan_range(
+            viewport.min_x()..viewport.max_x(),
+            target.min_x()..target.max_x(),
+        )
+        .start;
+        let new_pos_y = compute_pan_range(
+            viewport.min_y()..viewport.max_y(),
+            target.min_y()..target.max_y(),
+        )
+        .start;
```

Comment:
So we prefer the `top-left` of the rectangle to go into view. That seems reasonable, but should probably be documented.

Reply:
I added in a comment on the pass function. I'll do a more thorough documentation pass later.